### PR TITLE
Ensure that [[Set]] operation does not reject for TypedArray objects

### PR DESCRIPTION
**Pull request by rerobika**
JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-objects.c [hunk 1]
```diff
@@ -531,12 +531,10 @@ ecma_op_object_find_own (ecma_value_t base_value, /**< base value */
       /* ES2015 9.4.5.4 */
       if (ecma_object_is_typedarray (object_p))
       {
-#if ENABLED (JERRY_ESNEXT)
```
Comment (zherczeg) [inline jerry-core/ecma/operations/ecma-objects.c:534]: I think typedarray can be enabled without esnext.

**Review by zherczeg — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/operations/ecma-objects.c (hunk 1 above)
Comment (rerobika) [reply jerry-core/ecma/operations/ecma-objects.c:534]: If it is, that's a huge problem.

Comment (rerobika) [reply jerry-core/ecma/operations/ecma-objects.c:534]: No you cannot. `config.h:588` requires JERRY_ESNEXT for JERRY_BUILTIN_TYPEDARRAY

Comment (ossy-szeged) [reply jerry-core/ecma/operations/ecma-objects.c:534]: As far as I remember we don't support any ES5.1+ feature without ESNEXT enabled officially. Maybe it builds and works fine, but I think it isn't tested at all. If we would like to support eg TypedArray with ES5.1 profile, this configuration should have at least a buildoption test.

Comment (rerobika) [reply jerry-core/ecma/operations/ecma-objects.c:534]: No, the build does not work either.

Comment (zherczeg) [reply jerry-core/ecma/operations/ecma-objects.c:534]: Ok.

**Review by zherczeg — APPROVED** [review]
LGTM

**Review by dbatyai — APPROVED** [review]
LGTM

**Pull request merged**